site/mex_invenio/custom_facets.py

```diff
@@ -0,0 +1,36 @@
+from invenio_records_resources.services.records.facets import NestedTermsFacet
+
+class RestrictedTermsFacet(NestedTermsFacet):
+    """Facet that only displays specific values."""
+
+    def __init__(self, field, label, allowed_values=None, **kwargs):
+        """Initialize facet with allowed values."""
+        super().__init__(field=field, label=label, **kwargs)
+        self.allowed_values = set(allowed_values) if allowed_values else None
+
+    def get_labelled_values(self, data, filter_values, bucket_label=True, key_prefix=None):
+        """Get an unlabelled version of the bucket, filtered by allowed values."""
+        out = []
+        label_map = self.get_label_mapping(data.buckets)
+        for bucket in data.buckets:
+            key = full_key = self.get_value(bucket)
+            if key_prefix:
+                full_key = key_prefix + self._splitchar + full_key
```

Comment:
This can be moved under the allowed_values check on line 20.  Otherwise, full_key is set but never used.